vms/platformvm/state/stakers.go

```diff
@@ -289,6 +289,7 @@ func (s *diffStakers) PutValidator(staker *Staker) {
 		s.addedStakers = btree.NewG(defaultTreeDegree, (*Staker).Less)
 	}
 	s.addedStakers.ReplaceOrInsert(staker)
+	delete(s.deletedStakers, staker.TxID)
```

Comment:
when adding a validator, we should not mark it as deleted if it was previously deleted. This allows, for instance, to freely delete a non-yet-inserted staker before inserting it.
Note that "delete then add staker" does not currently happen in our system, but it may in the near future. Also "delete then add staker" is perfectly fine with state.State.
This change makes diff more aligned with state.State behaviour